### PR TITLE
File#content enhancements

### PR DESCRIPTION
**Pull request by vmoravec**
## Edit:

To avoid the use of Tempfile and better data stream handling now we can access the chunked body of the response directly from within the block parameter (thanks to jreidinger for the hint):

``` ruby
# setup the StudioApi::Connection
remote_file = StudioApi::File.find 123
remote_file.content do |response|
  # write the chunked response somwhere
  File.open('some_file','w') {|f| f.write response }
end
```

The same is valid for the Rpm#content method. Test for both classes added.
## Orig

I added a block to the #content method to preserve the current functionality (returning the http response directly) and to avoid loading the files into memory. When the block is used the file is being saved as a Tempfile first and then it is to be accessed from the block parameter for further manipulation.
## Example:

``` ruby
# setup the StudioApi::Connection
# find the file
remote_file = StudioApi::File.find 123
destination = '/home/user/new_file'

remote_file.content do |temp_file|
  # do something with the temp_file
  File.copy temp_file.path, destination
end
# the Tempfile is deleted at the end of the block
```
## Other:
- missing tests: the Fakeweb gem is a beast for testing the functionality of that method, some hint please so I could make some testing here

### CODE REVIEW

**Comment by jreidinger** [comment]
Well, fakeweb is beast for such case, but I think we should keep consistency of test suite. For example test of download see https://github.com/jreidinger/studio_api/blob/master/test/rpm_test.rb#L42
So please add test case, Thanks

File: lib/studio_api/generic_request.rb [hunk 1]
```diff
@@ -102,18 +113,31 @@ def do_request(request)
...
-        end
+        handle_active_resource_exception response
         response.body
       end
     end
 
+    def do_tempfile_request request, &block
+      request.basic_auth @connection.user, @connection.password
+      @http.start do |http|
+        Tempfile.open SecureRandom.hex(10) do |tmp|
```
Comment (jreidinger) [inline lib/studio_api/generic_request.rb]: What is purpose of secure random? TempFile itself should ensure uniqueness of file and proper access rights.

**Comment by jreidinger** [comment]
Otherwise commit looks good, partial reading is fine and maybe you can extend it also to rpm download.

File: lib/studio_api/generic_request.rb (hunk 1 above)
Comment (vmoravec) [inline lib/studio_api/generic_request.rb]: Tempfile::open method requires an argument; the idea was to improve the randomnes of the file name. By default there are being generated 6 random characters at the end of the file. If this is enough I'll remove the securerandom part and put there an empty string or maybe a rand(10).to_s or similar?

Comment (jreidinger) [inline lib/studio_api/generic_request.rb]: yes, removing dependency on secure randon is enough

File: lib/studio_api/file.rb [hunk 2]
```diff
@@ -28,9 +28,13 @@ class File < ActiveResource::Base
 
     # Downloads file to output. Allow downloading to stream or to path.
     # @return [String] content of file
```
Comment (jreidinger) [inline lib/studio_api/file.rb]: There is also change, that return value is changed to content of file or return value of block

**Pull request merged**